### ISSUE TITLE
validateIcons raises MANIFEST_ICON_NOT_FOUND on Windows even if icon exists

### ISSUE THREAD

**Issue by khris** [mozilla/addons-linter#1431]
### Describe the problem and steps to reproduce it:

`MANIFEST_ICON_NOT_FOUND` is raised when run `web-ext lint`.

1. Set icon path with files in sub-directory.
```json
  "icons": {
    "48": "icons/icon.png",
    "96": "icons/icon@2x.png"
  },

```
2. Run `web-ext lint` in working directory on Windows.
3. MANIFEST_ICON_NOT_FOUND

### What happened?

`MANIFEST_ICON_NOT_FOUND` even if icons exists.

### What did you expect to happen?

No error on Windows too.

### Anything else we should know?

It's because of path separator like #1426.

https://github.com/mozilla/addons-linter/blob/517b98b21cdff4fd850a5f713153506c4cdf92cd/src/parsers/manifestjson.js#L166 :
```javascript
if (!this.io.files.hasOwnProperty(path)) {
```

`this.io.files` has path like `icons\icon.png` not `icons/icon.png` when running on Windows. It causes `MANIFEST_ICON_NOT_FOUND`.

I tried fixing it and sending PR but it's not small to fix alone. `io.files`(it means both return value of `IOBase.getFiles()` and variables named `files` in some files) is used widely without any wrapping method. It seems to need design decision. In my opinion, paths should be stored as something other instead of plain string joinned with `path.join()`.

**Comment by kumar303** [comment]
^ It looks like a lot of `web-ext` users are running into this

**Comment by EnTeQuAk** [comment]
I'll take a look at this. I thought this might have been fixed with https://github.com/mozilla/addons-linter/pull/1432 but it doesn't look like it.

**Comment by EnTeQuAk** [comment]
There's also https://github.com/mozilla/addons-linter/pull/1432 blocking windows compatibility partially.

**Comment by muffinresearch** [comment]
> this.io.files has path like icons\icon.png not icons/icon.png when running on Windows. It causes MANIFEST_ICON_NOT_FOUND.

This is probably complicated by looking at dirs vs working with a zip which are different code paths. We'll probably want to unify everything on `/` as a path separator.

**Comment by asamuzaK** [comment]
I think this should fix the issue.

```
const { pathname } = new URL(iconPath, 'https://example.com/');
return path.join(pathname.split('/'));
```

[addons-linter/manifestjson.js at 9fa5bdd80fd51548e2272a17d0157f92fe20b153 · mstriemer/addons-linter](https://github.com/mstriemer/addons-linter/blob/9fa5bdd80fd51548e2272a17d0157f92fe20b153/src/parsers/manifestjson.js#L20 "addons-linter/manifestjson.js at 9fa5bdd80fd51548e2272a17d0157f92fe20b153 · mstriemer/addons-linter")

**Comment by asamuzaK** [comment]
Filed a PR #1440

**Comment by sneakypete81** [comment]
This doesn't just affect icons, I see MANIFEST_BACKGROUND_FILE_NOT_FOUND too.

Web-ext has been broken for Windows users for a while now, so it would be great to see this fixed.

**Comment by EnTeQuAk** [comment]
This should have been fixed by https://github.com/mozilla/addons-linter/pull/1730 and a few other PRs which add windows compatibility and testing.

In case it's still an issue, please let us know and feel free to re-open this issue at any time :)

**Comment by vcarciu** [comment]
Verified in Windows10x64 , using this [webext](https://github.com/mozilla/addons-linter/files/1181863/webextensions_test-0.0.1.zip) : 
![addonslinter](https://user-images.githubusercontent.com/1583842/34834952-160c64ce-f6fc-11e7-9df8-3f4b1af67014.png)
